drivers/w1/w1_ds2484.c

```diff
@@ -0,0 +1,181 @@
+/*
+ * Copyright (c) 2022 Caspar Friedrich <c.s.w.friedrich@gmail.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "w1_ds248x.h"
+
+#include <zephyr/devicetree.h>
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/drivers/i2c.h>
+#include <zephyr/drivers/w1.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/pm/device.h>
+#include <zephyr/zephyr.h>
+
+#define DT_DRV_COMPAT maxim_ds2484
+
+LOG_MODULE_REGISTER(ds2484, CONFIG_W1_LOG_LEVEL);
+
+struct ds2484_config {
+	struct w1_master_config w1_config;
+	const struct i2c_dt_spec i2c_spec;
+	const struct gpio_dt_spec slpz_spec;
+	bool apu;
+};
+
+struct ds2484_data {
+	struct w1_master_data w1_data;
+	uint8_t reg_device_config;
+};
+
+static int ds2484_reset_bus(const struct device *dev)
```

Comment:
Should these set of functions be `static inline`?

Reply:
@galak These are the API functions, do you mean to explicitly inline the ds248x_reset_bus et al. called from here?
If yes, I do agree.

Reply:
yeah, the ds248x_reset_bus versions.

Reply:
The header look quite weird then, but impact on build system is reduced this way. Done.